test/unit/cluster_resource_discovery_test.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+require 'test_helper'
+
+class ClusterResourceDiscoveryTest < KubernetesDeploy::TestCase
+  def test_global_resource_kinds_failure
+    crd = mocked_cluster_resource_discovery(nil, success: false)
+    kinds = crd.global_resource_kinds
+    assert_equal(kinds, [])
+  end
+
+  def test_global_resource_kinds_success
+    crd = mocked_cluster_resource_discovery(full_response)
+    kinds = crd.global_resource_kinds
+    %w(MutatingWebhookConfiguration ComponentStatus ComponentStatus).each do |kind|
```

Comment:
The second and third are the same, presumably not intentional? I'd also suggest adding an assertion on the total number of kinds discovered.